### DISCUSSION TITLE
sktime - HIVECOTEV2 model parameters issue

### DISCUSSION THREAD

**Discussion by monasound** [alan-turing-institute/sktime#5173] [Q&A] [answered]
When I try to run the HIVECOTEV2 model, the following is the error that I encounter. I am using default parameters.  I dont understand what the error is and I appreciate your comments.

from sktime.classification.hybrid import HIVECOTEV2
hc2 = HIVECOTEV2(time_limit_in_minutes=1,random_state=1234)
hc2.fit(X_train_Lullymore2 ,y_train_Lullymore)

**UserWarning: TemporalDictionaryEnsemble warning: min_window = 10 is larger than max_window = 9. min_window has been set to 9.
  warnings.warn(**

**Comment by fkiraly** [comment, marked as answer]
It looks like this comes from an interaction of TDE's defaults `min_window` and `max_window_prop` parameters, and your time series being - probably? - of length 9.

The default for `max_window` is length of your time series (instances), which I assume is 9. And for `min_window` it is the integer 10.

That should explain the warning?

As a side note, if your time series are very short, you may like to try simply flattening to an n times 9 data frame (instances times timepoits), and applying any `sklearn` classifier (e.g., `RandomForestClassifier`) - and use this as a "simple tabular" baseline.

Why: many of the "bespoke" classifiers assume long and/or reasonably smooth time series. Wheras 9 time points usually means very coarse sampling.

The code for a "tabularized" classifier, with `sktime` interface, would be:
```python
from sklearn.ensemble import RandomForestClassifier
from sktime.transformations.panel.reduce import Tabularizer

clf = Tabularizer() * RandomForestClassifier()
```

**Reply by monasound** [reply]
Yes, the one that I use consist of 9. I appreciate your help very much. Thank you for your time.

**Reply by fkiraly** [reply]
you're welcome! If this solved your problem, can you click "mark as answer" so this question is filed as answered?

**Reply by fkiraly** [reply]
some additional info:

* in `HIVECOTEV2` you can set the parameters of the wrapped TDE via `tde_params`
* you can build your own HIVE-COTE-like ensembles using `WeightedEnsembleClassifier` out of your favourite classifiers, if one of the components in the bespoke ensembles (like HIVECOTEVXYZ) keep acting up

**Reply by monasound** [reply]
Thank you once again
